packages/model-registry/upstream/frontend/src/odh/components/ModelCatalogDeployModalExtension.tsx

```diff
@@ -20,75 +25,66 @@ const ModelCatalogDeployModalExtension: React.FC<ModelCatalogDeployModalExtensio
   model,
   render,
 }) => {
-  const navigate = useNavigate();
-  const [extensions, extensionsLoaded] = useResolvedExtensions(isModelCatalogDeployModalExtension);
-
-  const [openModal, setOpenModal] = React.useState(false);
-
-  const [availablePlatformIds, setAvailablePlatformIds] = React.useState<string[]>([]);
-  const buttonState = getDeployButtonState(availablePlatformIds, true);
-
-  const onOpenModal = React.useCallback(() => {
-    setOpenModal(true);
-  }, [setOpenModal]);
-
-  const isModalAvailable = React.useMemo(
-    () => extensionsLoaded && extensions.length > 0,
-    [extensionsLoaded, extensions],
-  );
-
   const params = useParams<CatalogModelDetailsParams>();
   const decodedParams = decodeParams(params);
-  const [artifacts, artifactLoaded, artifactsLoadError] = useCatalogModelArtifacts(
+  const [artifacts] = useCatalogModelArtifacts(
```

Comment:
Removing `artifactLoaded` and `artifactsLoadError` here means we won't show a loading spinner or handle errors if the artifacts fail to load. I guess that's ok since the model details page where this is rendered will already handle that, but I wonder if what we should actually do is not bother fetching artifacts here, and instead pass the artifacts (or just the uri) down from the catalog page where we already have them loaded (ModelDetailsPage -> ModelCatalogDeployButton -> ModelCatalogDeployModalExtension).

Either way, we should probably have the deploy button be disabled if the artifacts are still loading or there was an error loading them.